ql/src/java/org/apache/hadoop/hive/ql/lockmgr/DbTxnManager.java

```diff
@@ -170,6 +170,7 @@ Licensed to the Apache Software Foundation (ASF) under one
   private ScheduledFuture<?> heartbeatTask = null;
   private static final int SHUTDOWN_HOOK_PRIORITY = 0;
   private final ReentrantLock heartbeatTaskLock = new ReentrantLock();
+  private String dbUnderReplication = null;
```

Comment:
Why do you need to store it?

Reply:
This variable is used by commitTxn/AbortTxn to know that this is hiveReplTxn and requires some special handling. It would be again set to null only when commit/AbortTxn event comes for this opened transaction.